### ISSUE TITLE
Investigate MicroProfile OpenAPI 2.0

### ISSUE THREAD

**Issue by jmini** [smallrye/smallrye-open-api#56]
Now that we have a first milestone release of MP-OpenAPI 2.0 ([2.0-MR1](https://github.com/eclipse/microprofile-open-api/releases/tag/mp-openapi-2.0-MR1)), I think it would make sense to create a `2.0` branch in this repo to work on an implementation that is compatible with this version.

I have started to do it to validate some other stuff I am working on. This is interesting because it allows to find some issue/limitation in the API.

Can a `2.0` branch be created in this repo?

This would be similar to #44 (for `2.0` instead of `1.1`)

**Comment by EricWittmann** [comment]
Branch `2.0` created.

**Comment by kenfinnigan** [comment]
There's already a `1.1` branch, would it be better to just move `master` to be 2.x?

**Comment by EricWittmann** [comment]
That's a really great point, considering we likely won't be making changes to the 1.0.x line anymore.

**Comment by EricWittmann** [comment]
Although maybe it makes more sense to create a `2.0` branch and move the `1.1` branch to master...

What do you guys think @jmini and @msavy ?

**Comment by kenfinnigan** [comment]
I don't think we want to maintain multiple branches of active development, so if there's going to be an MP OpenAPI 2.0 soon, then that should be what is developed on master.

**Comment by EricWittmann** [comment]
Side question, @kenfinnigan : Can we do releases of smallrye off any branch?  If yes, then having `master` === 2.0 probably makes sense.

**Comment by EricWittmann** [comment]
Ah OK - so here's the state of things.  We're going to be releasing version `1.1` of the specification this month.  Once that happens, we'll need to update SmallRye to support that version of the spec.  That is the purpose of the `1.1` branch here in SmallRye.  When we do that, `smallrye-open-api` will go from version `1.0.4` to `1.1.0`.

Far off in the distant future of 2019, we will be releasing version `2.0` of the specification (with more extensive changes).  At that time, we'll need to update SmallRye to implement the new spec, and presumably change `smallrye-open-api` to version `2.0.0`.

@jmini has requested a `2.0` branch here in SmallRye so that he can start making some changes to align with what's going on in spec 2.0, and I assume to evaluate those changes in a practical way.  So the 2.0 work is very preliminary, and based off a very early milestone release of the spec.  On the other hand, the 1.1 work is close to release.

**Comment by kenfinnigan** [comment]
Ok, in that case I would say it makes sense for there to be a `2.x` branch now, and the `1.1` branch is moved into master.

We only really need a branch when absolutely needed. Right now we don't need a branch for 1.0.x as we can always take one from a tag later.

**Comment by jmini** [comment]
No they will be a `1.1` release soon and this will be included in the next MP release (I have no idea when this will be)
See: https://github.com/eclipse/microprofile/issues/68

There is no ETA for `2.0` yet, this is work in progress.

---

I am already investing on `2.0` even if this is a Milestone release, because some changes made on this branch are required by some use case I have (which are not related to App server at all, but that might be interesting for people writing java tools for OpenAPI). 

I have decided to update Smallrye to `2.0` because I am using it to cross-verify that what I do also work with implementation I did not write myself. I think this is interesting for everybody.

If this is too early for you (I can understand that you do not want to maintain multiple versions), this is OK for me. I will continue to work on it on my fork.

I think that having the work on `2.0` in this project might be better if user people are interested (they have a chance to find it).

Just let me know.

**Comment by kenfinnigan** [comment]
I'm fine with a `2.x` branch for future work.

I was originally thinking it should be `master`, before I knew there was a `1.1` coming soon

**Comment by EricWittmann** [comment]
OK I think we have consensus.  I will do the following:

* Merge the `1.1` branch onto `master`
* Delete the current `2.0` branch that I created from the current master
* Re-create a `2.0` branch based on the new `master`

**Comment by kenfinnigan** [comment]
One nit from me on this, can we make it `2.x`?

**Comment by EricWittmann** [comment]
Yes. :)

**Comment by EricWittmann** [comment]
OK sounds good.

As for the `2.0` work - once I get the branch created (just waiting on approval of my 1.1 merge) you can submit whatever you have for `2.x` and I can clean it up from there.

**Comment by EricWittmann** [comment]
OK @jmini - the `1.1` branch has been merged onto `master` and there is now a new `2.x` branch.  I haven't done anything to the `2.x` branch except change the version in all the `pom.xml` files to `2.0.0-SNAPSHOT`.
